glnx-fdio.h

```diff
@@ -296,6 +296,51 @@ glnx_fstatat (int           dfd,
 }
 
 /**
+ * GLnxOptionalStat:
+ * @exists: Set to `TRUE` iff file exists
+ * @stbuf: Stat buffer
+ *
+ * Used by glnx_fstatat_or_noent().
+ */
+typedef struct {
+  gboolean exists;
+  struct stat stbuf;
+} GLnxOptionalStat;
+
+/**
+ * glnx_fstatat_or_noent:
+ * @dfd: Directory FD to stat beneath
+ * @path: Path to stat beneath @dfd
+ * @buf: (out caller-allocates): Return location for stat details
+ * @flags: Flags to pass to fstatat()
+ * @error: Return location for a #GError, or %NULL
+ *
+ * Like glnx_fstatat(), but handles `ENOENT` in a non-error way; this function
+ * returns a `GLnxOptionalStat` that also says whether or not the file exists.
+ * One could think of this as Option<struct stat> in Rust terms.
+ *
+ * Returns: %TRUE on success, %FALSE otherwise
+ * Since: UNRELEASED
+ */
+static inline gboolean
+glnx_fstatat_or_noent (int               dfd,
```

Comment:
Bikeshed: I find the `allow_noent` naming more descriptive, though whichever you prefer is fine!